cli/tsc/dts/lib.deno.unstable.d.ts

```diff
@@ -139,6 +141,159 @@ declare namespace Deno {
       notes?: MessageNote[];
     }
 
+    export interface Plugin {
+      name: string;
+      setup: (build: PluginBuild) => void | Promise<void>;
+    }
+
+    export interface PluginBuild {
```

Comment:
Would love to compare recent plugin apps before we settle on exposing the esbuild API as is. Both too and vite have recently updated their API to integrate many of the advancements esbuild made. And they do support the notion of multiple environments at once.